.github/workflows/validate.yml

```diff
@@ -1,26 +1,30 @@
+name: Validate
+
 on:
   push:
-    branches:
-      - master
+    branches: [master]
   pull_request:
-  release:
-    types:
-      - created
-name: Validate
+    branches: [master]
+
 jobs:
-  Go:
+  validate:
     strategy:
       matrix:
-        go-version: [1.13.x, 1.14.x, 1.15.x]
+        go-version: [1.14.x, 1.15.x]
         os: [ubuntu-18.04]
     runs-on: ${{ matrix.os }}
+    
     steps:
     - name: Install Go
       uses: actions/setup-go@v2
       with:
         go-version: ${{ matrix.go-version }}
-    - name: Checkout code
+
+    - name: Checkout Branch
       uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
With this, the PR checker can no longer guard against a failed merge. Also, `event.pull_request` is not available on push events.

 I suggest keeping the default checkout action, as it handles both cases as expected.

Reply:
Understood. I've created a new action, validate-merge, to handle the case of a failed merge and repurposed this one for branch code only. My intention with separating these out is to clearly express to the contributor why a build is failing. @hhhjort was understandably confused in a previous PR why his tests passed locally but failed on the server. Took us all a bit to figure it out, especially since Travis behaves differently.

The validate-merge task is also simpler since it just runs on one Go version (the prod one) and omits race detection and code coverage reports.

Reply:
> Also, event.pull_request is not available on push events.

Correct. I suspected the ref would resolve to an empty string if not present and default to the HEAD for push requests. I just ran a series of experiments which proved that's what is happening.